zulipterminal/ui_tools/views.py

```diff
@@ -967,27 +968,45 @@ def __init__(self, controller: Any, msg: Any) -> None:
         super().__init__(controller, self.widgets, 'MSG_INFO')
 
     def build_pile(self, snapshot: Dict[str, Any], tag: str='',
-                   ) -> Any:
+                   author_prefix: str='Edited') -> Any:
         timestamp = time.ctime(snapshot['timestamp'])[:-5]
+        author = self.get_author(snapshot['user_id'])
         topic = snapshot['topic']
 
         if 'prev_topic' in snapshot and topic != snapshot['prev_topic']:
             topic += ' (updated from {})'.format(snapshot['prev_topic'])
+            if 'prev_content' in snapshot:
+                author_prefix += ' and Topic updated'
+            else:
+                author_prefix = 'Topic updated'
 
         header = urwid.Columns([
                 urwid.Text(('edit_topic', topic)),
                 (19, urwid.Text(('edit_time', timestamp))),
             ], dividechars=2,
         )
         subheader = urwid.Columns([
+                urwid.Text(('edit_author',
+                            '{} by {}'.format(author_prefix, author))),
                 (18, urwid.Text(('edit_tag', '({} Version)'.format(tag) if tag
-                                else ''))),
+                                else ''), align='right')),
             ], dividechars=2,
         )
         message = urwid.Text(snapshot['content'])
         return urwid.Pile([urwid.AttrWrap(header, 'popup_contrast'), subheader,
                            message])
 
+    def get_author(self, user_id: int) -> str:
+        if self.msg['sender_id'] == user_id:
+            return self.msg['sender_full_name']
+        else:
+            for user in self.controller.view.users:
```

Comment:
While this does take `O(n)` in the worst case, it does if for every `edit` in edit-history. I was brainstorming something faster and maybe we can use a combination of `model.user_id_email_dict` and `model.user_dict` to get the name faster. Worthwhile to add a `FIXME` in case we decide to go with this for now.

Reply:
Thanks so much, the proposed strategy is faster than the `O(n)` approach.